services/compute/virtualmachine/wssd.go

```diff
@@ -305,3 +310,22 @@ func getComputeTags(tags *wssdcloudproto.Tags) map[string]*string {
 func getWssdTags(tags map[string]*string) *wssdcloudproto.Tags {
 	return prototags.MapToProto(tags)
 }
+
+func (c *client) virtualMachineValidations(opType wssdcloudproto.Operation, vmss *compute.VirtualMachine) error {
```

Comment:
We need to check for `OsProfile` nil value, right?

I don't think there's anything stopping from nil osconfig value here.

Reply:
yes, good catch

Reply:
nit: we won't come here if vmss is nil, I think we can avoid that for readability.

Reply:
done!